support-lib/wasm/djinni_wasm.hpp

```diff
@@ -602,6 +602,30 @@ class GenericBuffer: public DataObject {
 extern "C" void djinni_register_name_in_ns(const char* prefixedName, const char* namespacedName);
 extern "C" void djinni_throw_native_exception(const std::exception& e);
 
+template<typename U>
+struct DefaultInit {
+    static U get() { return U{}; }
+};
+template<>
+struct DefaultInit<em::val> {
+    static em::val get() { return em::val::undefined(); }
+};
+
+template<typename T>
+struct ExceptionHandlingTraits {
+    using R = typename T::JsType;
+    static R handleNativeException(const std::exception& e) {
+        djinni_throw_native_exception(e);
+        return DefaultInit<R>::get();
```

Comment:
Hmm I'm a bit confused about this line. For non-async functions, we used to have code like:
```
} catch (const std::exception& e) {
        djinni::djinni_throw_native_exception(e); 
        throw;
}
```

I think we still want that re-throw for non-native exceptions for non-async functions right?

Reply:
`djinni::djinni_throw_native_exception(e); ` throws a JS exception and will not return. 
The line after that was just to satisfy the compiler.

Reply:
Ahh got it, thanks https://cs.github.com/Snapchat/djinni/blob/5accefb308e9b958b67b2e533e5b0dc2d30eef4e/src/source/WasmGenerator.scala?q=djinni_throw_native_exception#L378-L381